src/libraries/System.Text.Json/src/System.Text.Json.csproj

```diff
@@ -14,8 +14,8 @@
   <PropertyGroup>
     <!-- For the inbox library (that is shipping with the product), this should always be true. -->
     <!-- BUILDING_INBOX_LIBRARY is only false when building the netstandard compatible NuGet package. -->
-    <DefineConstants Condition="$(TargetFramework.StartsWith('netcoreapp'))">$(DefineConstants);BUILDING_INBOX_LIBRARY</DefineConstants>
-    <NoWarn Condition="!$(TargetFramework.StartsWith('netcoreapp'))">$(NoWarn);nullable</NoWarn>   
+    <DefineConstants Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)' or '$(TargetFramework)' == 'netcoreapp3.0'">$(DefineConstants);BUILDING_INBOX_LIBRARY</DefineConstants>
```

Comment:
What about `netcoreapp3.1`?

Reply:
We don’t build System.Text.Json for netcoreapp3.1 and we shouldn’t need as it has been inbox since netcoreapp3.0.

Also it seems like we shouldn’t need to cross build System.Text.Json for netcoreapp3.0 since it is inbox and the package should just contain an inbox placeholder:


https://github.com/dotnet/runtime/blob/master/src/libraries/pkg/baseline/packageIndex.json#L6253

Reply:
We should do a pass over all our OOBs at some point and make sure that we only build what's actually necessary. This will help us keep our build configuration and conditions sane.